edu_edfi_airflow/dags/earthbeam_dag.py

```diff
@@ -501,6 +535,400 @@ def build_tenant_year_taskgroup(self,
+                )
+
+                task_order.append(python_preprocess)
+                paths_to_clean.append(airflow_util.xcom_pull_template(python_preprocess.task_id))
+
+            ## List Raw files for dynamic task mapping
+            list_files_task = PythonOperator(
+                task_id = f'{taskgroup_grain}_list_files_in_dir',
+                dag=self.dag,
+                python_callable=self.list_files_in_raw_dir,
+                op_kwargs={
+                    'raw_dir': airflow_util.xcom_pull_template(python_preprocess.task_id) if python_callable else raw_dir,
+                    'ds_nodash_string': '{{ ds_nodash }}',
+                    'ts_nodash_string': '{{ ts_nodash }}'
+                },
+                provide_context=True
+            )
+            task_order.append(list_files_task)
+
+            ### Raw to S3
+            if s3_conn_id:
+                if not s3_filepath:
+                    raise ValueError(
+                        "Argument `s3_filepath` must be defined to upload raw files to S3."
+                    )
+
+                s3_raw_filepath = edfi_api_client.url_join(
+                    s3_filepath, 'raw',
+                    tenant_code, self.run_type, api_year, grain_update,
+                    '{{ ds_nodash }}', '{{ ts_nodash }}'
+                )
+
+                raw_to_s3_kwargs = list_files_task.output.map(lambda raw_dir: {'local_filepath': raw_dir['local_filepath'] })
+                raw_to_s3 = DynamicPythonOperator.partial(
+                    task_id=f"{taskgroup_grain}_upload_raw_to_s3",
+                    python_callable=local_filepath_to_s3,
+                    static_kwargs={
+                        's3_conn_id': s3_conn_id,
+                        's3_destination_key': s3_raw_filepath,
+                        'remove_local_filepath': False,
+                    },
+                    provide_context=True,
+                    pool=self.pool,
+                    dag=self.dag,
+                    sla=None
+                ).expand(op_kwargs=raw_to_s3_kwargs)
+
+                task_order.append(raw_to_s3)
+
+
+            ### EarthmoverOperator: Required
+            em_output_dir = edfi_api_client.url_join(
+                self.em_output_directory,
+                tenant_code, self.run_type, api_year, grain_update,
+            )
+
+            em_state_file = edfi_api_client.url_join(
+                self.emlb_state_directory,
+                tenant_code, self.run_type, api_year, grain_update,
+                'earthmover.csv'
+            )
+
+            run_earthmover_kwargs = list_files_task.output.map(lambda input_file:
+                evaluate_kwargs(str(earthmover_kwargs),
+                    context={
+                        'input_file': input_file['local_filepath'],
+                        'output_dir': edfi_api_client.url_join(
+                            em_output_dir,
+                            input_file['ds_nodash_string'],
```

Comment:
Why are we passing `ds_nodash` and `ts_nodash` strings across these methods? These date stamps are already added when a user uses `EarthbeamDAG.build_local_raw_dir()`.

Reply:
can you help me with this? I'm not sure how to correctly construct output_dir here